controllers/apps/configuration/configuration_controller.go

```diff
@@ -151,12 +151,21 @@ func (r *ConfigurationReconciler) runTasks(taskCtx TaskContext, tasks []Task) (e
 	if err != nil {
 		return err
 	}
-
+	var dependencyObjs []client.Object
+	// HACK for hostNetwork
+	// TODO: define the api to inject dynamic info of the cluster components
+	if synthesizedComp.PodSpec.HostNetwork {
```

Comment:
It may be better to set the internal variables of SynthesizedComponent in component.buildSynthesizedComponent, so that dependencyObjs do not need to be generated in different code blocks.